README.md

```diff
@@ -1,333 +1,51 @@
-README for Mbed TLS
-===================
+This repository is Zephyr's fork of [Mbed TLS](https://github.com/Mbed-TLS/mbedtls).
 
-Mbed TLS is a C library that implements cryptographic primitives, X.509 certificate manipulation and the SSL/TLS and DTLS protocols. Its small code footprint makes it suitable for embedded systems.
+Additional patches
+------------------
 
-Mbed TLS includes a reference implementation of the [PSA Cryptography API](#psa-cryptography-api). This is currently a preview for evaluation purposes only.
+On top of upstream releases, additional patches are applied to this repository as necessary.
+<br>
+The additional patches are marked with different tags depending on their type.
+<br>
+For every one of them, their commit title is preprended with `[zep <patch type>]`.
+<br>
+Below is a description of the various patch types.
 
-Configuration
--------------
+### `fromtree`
 
-Mbed TLS should build out of the box on most systems. Some platform specific options are available in the fully documented configuration file `include/mbedtls/mbedtls_config.h`, which is also the place where features can be selected. This file can be edited manually, or in a more programmatic way using the Python 3 script `scripts/config.py` (use `--help` for usage instructions).
+This is the most common and preferred patch type.
+<br>
+A `fromtree` is simply a direct, clean cherry pick of an upstream commit.
 
-Compiler options can be set using conventional environment variables such as `CC` and `CFLAGS` when using the Make and CMake build system (see below).
+It shall be cherry picked with the `-x` flag so that the hash of the upstream commit is present in the commit message for tracking purposes.
+However, no sign-off shall be added by the person cherry picking the commit and the original commit message shall be left unmodified, only the commit title edited to preprend `[zep fromtree]` to it.
 
-We provide some non-standard configurations focused on specific use cases in the `configs/` directory. You can read more about those in `configs/README.txt`
+### `fromlist`
 
-Documentation
--------------
+This is equivalent to `fromtree`, except that `fromlist` is for upstream commits that haven't been merged yet.
 
-The main Mbed TLS documentation is available via [ReadTheDocs](https://mbed-tls.readthedocs.io/).
+We have an upstream-first approach, so any changes that can be upstream should first be submitted upstream.
+A `fromlist` comes in handy for cherry picking changes that have been submitted but not yet merged upstream.
+They should however have reasonable expectations to be merged with few modifications upstream, and it is preferable to get at least a first round of reviews on the upstream PR before cherry picking it to ensure that.
```

Comment:
I might be worth noting that 'fromlist' is historical, and this applies to changes living as PRs. It might be worth spelling out that Gerrit review is also a possibility.